src/drivers/serac.cpp

```diff
@@ -75,46 +75,48 @@ int main(int argc, char *argv[])
-                 "Relative tolerance for the Newton solve.");
-  args.AddOption(&nonlin_params.abs_tol, "-nabs", "--newton-absolute-tolerance",
-                 "Absolute tolerance for the Newton solve.");
-  args.AddOption(&nonlin_params.max_iter, "-nit", "--newton-iterations", "Maximum iterations for the Newton solve.");
-  args.AddOption(&nonlin_params.print_level, "-npl", "--newton-print-level", "Newton print level.");
-  args.AddOption(&t_final, "-tf", "--t-final", "Final time; start time is 0.");
-  args.AddOption(&dt, "-dt", "--time-step", "Time step.");
+  CLI::App app{"serac: a high order nonlinear thermomechanical simulation code"};
+  app.add_option("-m, --mesh", mesh_file,  "Mesh file to use.", true);
+  app.add_option("--rs, --refine-serial", ser_ref_levels,  "Number of times to refine the mesh uniformly in serial.", true);
+  app.add_option("--rp, --refine-parallel", par_ref_levels, 
+                  "Number of times to refine the mesh uniformly in parallel.", true);
+  app.add_option("-o, --order", order,  "Order degree of the finite elements.", true);
+  app.add_option("--mu, --shear-modulus", mu,  "Shear modulus in the Neo-Hookean hyperelastic model.", true);
+  app.add_option("-K, --bulk-modulus", K,  "Bulk modulus in the Neo-Hookean hyperelastic model.", true);
+  app.add_option("--tx, --traction-x", tx,  "Cantilever tip traction in the x direction.", true);
+  app.add_option("--ty, --traction-y", ty,  "Cantilever tip traction in the y direction.", true);
+  app.add_option("--tz, --traction-z", tz,  "Cantilever tip traction in the z direction.", true);
+  app.add_flag("--slu, --superlu, !--no-slu, !--no-superlu", slu_solver,  "Use the SuperLU Solver.");
+  app.add_option("--lrel, --linear-relative-tolerance", lin_params.rel_tol, 
+                  "Relative tolerance for the lienar solve.", true);
+  app.add_option("--labs, --linear-absolute-tolerance", lin_params.abs_tol, 
+                  "Absolute tolerance for the linear solve.", true);
+  app.add_option("--lit, --linear-iterations", lin_params.max_iter,  "Maximum iterations for the linear solve.", true);
+  app.add_option("--lpl, --linear-print-level", lin_params.print_level,  "Linear print level.", true);
+  app.add_option("--nrel, --newton-relative-tolerance", nonlin_params.rel_tol, 
+                  "Relative tolerance for the Newton solve.", true);
+  app.add_option("--nabs, --newton-absolute-tolerance", nonlin_params.abs_tol, 
+                  "Absolute tolerance for the Newton solve.", true);
+  app.add_option("--nit, --newton-iterations", nonlin_params.max_iter,  "Maximum iterations for the Newton solve.", true);
+  app.add_option("--npl, --newton-print-level", nonlin_params.print_level,  "Newton print level.", true);
+  app.add_option("--dt, --time-step", dt,  "Time step.", true);
 
   // Parse the arguments and check if they are good
-  args.Parse();
-  if (!args.Good()) {
+  try {
```

Comment:
Real exception throws! Yay!

Reply:
We did discuss the possibility of adopting a Google C++ code style guide in the past, and their policy on exceptions is pretty severe:

`Do not use C++ exceptions.`

I have no objection to the use of exceptions here, but I think it might be worth restricting their use to the higher level parts of the code.

Reply:
Agreed that exceptions should only be allowed on this "driver" level. We should also probably start drafting a code style guide.

Reply:
Actually, this will probably get replaced by a SLIC call. See [this PR.](https://github.com/LLNL/serac/pull/78)

Reply:
A dumb question, but this will be "thrown" if there is an undefined command line argument correct? On my other project I know we have waffled back and forth on this because it "insta-kills" batch jobs. 

I'm wondering if there is a way we can "trial-run-test" our command-line options without actually running anything?